crates/js-sys/src/lib.rs

```diff
@@ -478,6 +478,9 @@ extern "C" {
     pub fn slice_with_end(this: &SharedArrayBuffer, begin: u32, end: u32) -> SharedArrayBuffer;
 }
 
+unsafe impl Send for SharedArrayBuffer {}
```

Comment:
@alexcrichton is it fine to mark `SharedArrayBuffer` as `Send + Sync` manually? I tried to remove `#[wasm_bindgen(extends = Object)]`, but the compiler still thought that `SharedArrayBuffer` had a `PhantomData` from `JsValue`

Reply:
Unfortunately this can't be done because the `JsValue` underlying can't actually be sent between threads (inherently). The `SharedArrayBuffer` value that you'll be notifying through will need to be `wasm_bindgen::memory()` I believe

Reply:
It's an interesting point. So I can remove `unsafe impl` from here?
I have some more questions about implementation.
- WebAssembly memory could be just an ArrayBuffer, not the shared one. Is it fine?
- In docs there is `pub fn memory() -> JsValue`, which refers to the whole linear memory. Do I need to use `std::alloc` here to obtain a proper memory location?

Reply:
Oh sorry forgot to follow-up but yes it should be safe to remove these.